test/test_type.jl

```diff
@@ -25,6 +25,39 @@ isdir(outdir) && rm(outdir, recursive = true)
         end
     end
 
+    @timed_testset "TreeMesh & SerialTree type consistence" begin
+        for RealT in (Float32, Float64)
+            coordinates_min = -convert(RealT, 1)
+            coordinates_max = convert(RealT, 1)
+
+            mesh = TreeMesh(coordinates_min, coordinates_max,
+                            initial_refinement_level = 6,
+                            n_cells_max = 30_000)
+
+            @test typeof(@inferred mesh.tree) == SerialTree{1, RealT}
```

Comment:
This does not really test what you wanted to fix, does it?

Reply:
Yes, true. 
I add now the helper function 
```julia
get_tree(mesh::TreeMesh) = mesh.tree
```

for which on main currently

```julia
julia> @inferred get_tree(mesh)
ERROR: return type Trixi.SerialTree{2, Float64} does not match inferred return type Trixi.SerialTree{2}
Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:35
 [2] top-level scope
   @ REPL[5]:1
```

gets returned.